extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/MapboxCarUtils.kt

```diff
@@ -0,0 +1,180 @@
+ * Fired when there's a map scroll event.
+ */
+@MapboxExperimental
+fun interface OnMapScrollListener {
+  /**
+   * Map scroll event fired.
+   */
+  fun onMapScroll()
+}
+
+/**
+ * Fired when there's a map scale event.
+ */
+@MapboxExperimental
+fun interface OnMapScaleListener {
+  /**
+   * Map scroll event fired.
+   */
+  fun onMapScale()
+}
+
+/**
+ * Init Mapbox Map surface for the car app.
+ *
+ * @param mapInitOptions options to initialise a MapboxMap
+ * @param scrollListener
+ * @param scaleListener
+ * @param mapSurfaceReadyCallback
+ */
+@MapboxExperimental
+fun Session.initMapSurface(
+  mapInitOptions: MapInitOptions = MapInitOptions(carContext),
+  scrollListener: OnMapScrollListener? = null,
+  scaleListener: OnMapScaleListener? = null,
+  mapSurfaceReadyCallback: MapSurfaceReadyCallback
+) {
+  var mapSurface: MapSurface? = null
+  val surfaceCallback: SurfaceCallback = object : SurfaceCallback {
+    override fun onSurfaceAvailable(surfaceContainer: SurfaceContainer) {
+      synchronized(this) {
+        Log.i(TAG, "Surface available $surfaceContainer")
+        surfaceContainer.surface?.let { surface ->
+          mapSurface = MapSurface(carContext, surface, mapInitOptions).also { surface ->
+            surface.surfaceCreated()
+            surface.surfaceChanged(surfaceContainer.width, surfaceContainer.height)
+            mapSurfaceReadyCallback.onMapSurfaceReady(surface)
+          }
+        }
+      }
+    }
+
+    override fun onVisibleAreaChanged(visibleArea: Rect) {
+      synchronized(this) {
+        Log.i(
+          TAG,
+          "Visible area changed $visibleArea"
+        )
+      }
+    }
+
+    override fun onStableAreaChanged(stableArea: Rect) {
+      synchronized(this) {
+        Log.i(
+          TAG,
+          "Stable area changed $stableArea"
+        )
+      }
+    }
+
+    override fun onSurfaceDestroyed(surfaceContainer: SurfaceContainer) {
+      synchronized(this) {
+        Log.i(TAG, "Surface destroyed")
+        mapSurface?.surfaceDestroyed()
+      }
+    }
+
+    override fun onScroll(distanceX: Float, distanceY: Float) {
+      Logger.e(TAG, "onScroll $distanceX, $distanceY")
+      synchronized(this) {
+        mapSurface?.onScroll(distanceX, distanceY)
+        scrollListener?.onMapScroll()
+      }
+    }
+
+    override fun onScale(focusX: Float, focusY: Float, scaleFactor: Float) {
+      Logger.e(TAG, "onScroll $focusX, $focusY, $scaleFactor")
+      synchronized(this) {
+        mapSurface?.onScale(focusX, focusY, scaleFactor)
+        scaleListener?.onMapScale()
+      }
+    }
+  }
+
+  lifecycle.addObserver(
+    object : DefaultLifecycleObserver {
+      override fun onCreate(owner: LifecycleOwner) {
+        Log.i(TAG, "SurfaceRenderer created")
+        carContext.getCarService(AppManager::class.java)
+          .setSurfaceCallback(surfaceCallback)
+      }
+
+      override fun onStart(owner: LifecycleOwner) {
+        Log.i(TAG, "onStart")
+        mapSurface?.onStart()
```

Comment:
Should be also wrapped with `synchronized(this)` as surface and surface callback could be accessed from different threads.